### ISSUE TITLE
"Efficient neural decoding of self-location with a deep recurrent network"

### ISSUE THREAD

**Issue by mathieuboudreau** [Notebook-Factory/assign#1]
Paper: https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006822
Code: https://github.com/NeuroCSUT/RatGPS
Note: Python 2, Jupyter Notebooks, reproduces some figures.

**Comment by mathieuboudreau** [comment]
# To-Do (Create Plotly Figures)

* [x] Figure 1 
  * Located in RatGPS/plots/article_plots.ipynb
  * [x] a)
  * [x] b)
* [x] Figure 2
  * Located in RatGPS/2D/results.ipynb
  * [x] a)
  * [x] b)
  * [x] c)
* [x] Figure 3
  * Located in RatGPS/plots/article_plots.ipynb
  * [x] a)
  * [x] b)
  * [x] c)
  * [x] d)
* [x] Figure 4
  * Located in RatGPS/2D/results.ipynb
  * [x] a)
  * [x] b)
  * [x] c)
    * Resources:
      * https://community.plotly.com/t/how-to-create-hexbin-plot/13275
      * https://nbviewer.jupyter.org/gist/empet/8e466955c9e30f7471b4fb45c3a0fb21
  * [x] d)
* [x] ~Figure 5~
  * Missing in notebooks
* [x] ~Figure 6~
  * Missing in notebooks

**Comment by mathieuboudreau** [comment]
# To-Do (Format figures like paper)

* [x] Figure 1 
  * Located in RatGPS/plots/article_plots.ipynb
  * [x] a)
    * No capital for "e" in error on the y label.
    * No capital for "w" in Window on the x label
    * No capital for "d" in decoder in legend
  * [x] b)
    * No capital for "e" in error on the y label.
    * No capital for "w" in Window on the x label
    * No capital for "d" in decoder in legend
* [x] Figure 2
  * Located in RatGPS/2D/results.ipynb
  * [x] a)
  * [x] b)
     * No capital for "e" in error on the y label.
     * No capital for "d" in decoder in legend.
  * [x] c)
     * No capital for "e" in error on the y label.
     * No capital for "d" in decoder in legend.
* [x] Figure 3
  * Located in RatGPS/plots/article_plots.ipynb
  * [x] For all of them:
     * [x] Change pink to dark red
     * [x] No capital for "e" in error on the y label.
     * [x] No capital for "m" in memory in legend.
  *  **To do later on, fix the data label size on each bar to be bigger, but not currently high priority**
* [x] Figure 4
  * Located in RatGPS/2D/results.ipynb
  * [x] a)
  * [x] b)
  * [x] c)
  * [x] d)
* [x] Figure 5
* [x] Figure 6
  * Remove figure 6 code and content, replace with "to do" or "work in progress"

**Comment by mathieuboudreau** [comment]
# To Do (Format Jupyter Book)

* [x] Intro
  * [x] Give full citation for paper and not just title, something like : "Welcome to an interactive Jupyter Book for the research paper: Larremore DB (2019) Bayes-optimal estimation of overlap between populations of fixed size. PLOS Computational Biology 15(3): e1006898." **BUT FOR YOUR PAPER CITATION**
  * [x] Remove `et al.` 
  * [x] Start second paragraph with this:  "This Jupyter Book reproduces their paper figures, and lets you interact with them".
  * [x] Make "PLOS COMPUTATIONAL BIOLOGY" logo and put it at the bottom
  * [x] Remove current figure

* [x] Theory
  * [x] Remove
* [x] Remove "Interactive figures" section
* [x] Make all figures in the sidebar permanent (no collapse)
* [x] Remove all old static matplotlib figures & code
* [x] Hide all code above figures
* [x] Split code like this (with hidden cells)
  * Cell 1: Imports
  * Cell 2: functions
  * Cell 3: Whatever other code is needed to create the data for the figure
  * Cell 4: figure code
* [x] Add external link to the sidebar to the paper

**Comment by mathieuboudreau** [comment]
# Notes from June 30th meeting

To do:

* [x] Create figure(s) 4
* [x] Open issue about not being able to fix/choose version of sklearn in the Python 2 environment, tagging @TommyBoshkovski because it's all his fault we're using this Binder setup and not a Dockerfile.
* [x] ~Mayyyyybe write the hex plot script into a function, in case we want to do figures 5 and 6. But, don't waste 7 hours on this.~

**Comment by TommyBoshkovski** [comment]
If I understood correctly, you need a specific version of sklearn to be installed in the py2 environment? If that's the case just specify the version in the requirements.txt. :)

**Comment by zelenkastiot** [comment]
Hey guys, opened this issue https://github.com/Notebook-Factory/NeuralDecoding_book/issues/3 for the version of sklearn.

**Comment by agahkarakuzu** [comment]
@zelenkastiot I checked the book once again, nice job! What I noticed first is that your `css` interferes with some features:

1. The text is not aligned properly (at least on my window MBP 13"). On the left, it should align with the download button and on the right, sentences are cut. 

![image](https://user-images.githubusercontent.com/9632322/86607599-75683b00-bf77-11ea-9ce3-16fde688f959.png)

2. Sticky menu overlaps with the text:

![image](https://user-images.githubusercontent.com/9632322/86607854-c8da8900-bf77-11ea-9db6-2d121af6360a.png)

Interactive version of the Fig. 5 is amazing! Interactive hex density plots with dropdown 🎊

**Comment by zelenkastiot** [comment]
Thanks for the review @agahkarakuzu!! I will try to fix the <code> css </code> by tomorrow. I will also push some other features I have been working on / fixing :metal: :metal:

**Comment by zelenkastiot** [comment]
Hey everyone ( @mathieuboudreau, @agahkarakuzu , @TommyBoshkovski ) , I managed to modify almost everything 🥂 

Demo-Book : https://notebook-factory.github.io/NeuralDecoding_book/intro.html 

Changes: 
- Edited content
- Added **PLOS** on intro page (added a smaller logo and I modified the book logo)  
- Deleted plots from original code 
- Added <code>(View plot code)</code> tags on each hidden cell.

**Comment by agahkarakuzu** [comment]
Thanks @zelenkastiot, this looks good! 

I think we should not override the width of the `div.jb_cell`. Again, it is cut on the right border. When I edit it like this, the problem somewhat resolves, **still not ideal**. 

```diff
div.jb_cell {
-    width: 100%;
+   width: 60vw%;
}
```

The problem with the sticky navigation (TOC on the right) still persists. 

Aesthetic changes (color of the left pane, text style etc.) are ok, but I think we should retain the original layout. When you do that, some figures will be too large for the narrower column, you may need to make them smaller as well.

**Comment by zelenkastiot** [comment]
Thanks for the feedback @agahkarakuzu!

I will retain the original layout by tomorrow 😄

**Comment by zelenkastiot** [comment]
Hey @agahkarakuzu I pushed the changes we discussed this morning. Hope the layout is better now 😄 

Demo-Book : https://notebook-factory.github.io/NeuralDecoding_book/intro.html

**Comment by agahkarakuzu** [comment]
Hey @zelenkastiot, there are still issues :/

![image](https://user-images.githubusercontent.com/9632322/86957690-134f3780-c129-11ea-867d-b7eb1d0222f7.png)

This layout where figures are aligned with the text, not overflowing is much classy:

![image](https://user-images.githubusercontent.com/9632322/86957820-41347c00-c129-11ea-9e7a-46efdc6bc2cc.png)


Can we make it so that we just have some style changes but the layout of the book repeats https://qmrlab.org/t1_book/01/ir_blog/IR_SignalModelling.html

**Comment by mathieuboudreau** [comment]
@agahkarakuzu I used HTML div properties to setup the paragraphs like that if I recall, there's no way to created full-justified with markdown-only. @zelenkastiot feel free to copy my HTML code - you'll need to add the cell from the bottom of my notebooks that formats the html of all the cells above them after they've been run.

**Comment by agahkarakuzu** [comment]
@mathieuboudreau I was referring to how figures were aligned with the text rather than how the text is aligned.

I am totally ok with left aligned text as long as the width of the figures does not exceed that of the text, all the content, buttons etc are visible.

**Comment by mathieuboudreau** [comment]
Ah ok - in that case I think his figures are just too wide - use my figure sizes for reference

**Comment by zelenkastiot** [comment]
Sure thing guys, I saw @mathieuboudreau's book, will fix it. Thanks 🤘

**Comment by mathieuboudreau** [comment]
Meeting to-do list:

* [ ] Look into a way to reduce space between code cells and figures or text
* [x] Explore different ways to visualize figures/code in a fork

**Comment by zelenkastiot** [comment]
Hey guys, 

Tried out a different layout in this fork: https://github.com/zelenkastiot/NeuralDecoding_book
Forked book: https://zelenkastiot.github.io/NeuralDecoding_book/intro.html


<br> 
Added the following features:

- Made all links to figures in [README](https://github.com/zelenkastiot/NeuralDecoding_book/blob/master/README.md) to launch binder with **jupyterlab**

- Added a <code> Toggle Sidebar</code>
![image](https://user-images.githubusercontent.com/37374247/89281959-1d425880-d64b-11ea-9c87-5db0153a67bf.png)

- Removed the right navigation pane 

- Modified the order of the cells 
1.**Figure heading**  
2.(python2 cell)
3.(plot code cell)  

![image](https://user-images.githubusercontent.com/37374247/89282447-d7d25b00-d64b-11ea-8721-785a46e13b22.png)

- Updated plotly to newest version 

- Added this <code> hovermode = 'x'</code> for Figure 1, think the hover is fancier this way
![image](https://user-images.githubusercontent.com/37374247/89282987-ac9c3b80-d64c-11ea-8156-e77529f0e391.png)


- Made the navigation pane on the bottom with black text instead of hyperlinked 

![image](https://user-images.githubusercontent.com/37374247/89283166-f9801200-d64c-11ea-88ce-3617bacfe625.png)

I may work around some other features until our call later 🤘

**Comment by zelenkastiot** [comment]
p.s. 

Also I tried to make the <code> custom.css </code> as much adaptive to different screen resolutions as possible (by using %, <code>em</code>). I checked it on multiple resolutions by copy - pasting the link of my page [here]( http://whatismyscreenresolution.net/multi-screen-test)
